.github/scripts/generate_binary_build_matrix.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+"""Generates a matrix to be utilized through github actions
+
+Will output a condensed version of the matrix if on a pull request that only
+includes the latest version of python we support built on three different
+architectures:
+    * CPU
+    * Latest CUDA
+    * Latest ROCM
+"""
+
+import json
+import os
+import itertools
+
+CUDA_ARCHES = [
+    "10.1",
+    "10.2",
+    "11.0"
+]
+
+ROCM_ARCHES = [
+    "3.10",
+    "4.0"
+]
+
+FULL_ARCHES = [
+    "cpu",
+    *CUDA_ARCHES,
+    *ROCM_ARCHES
+]
+
+CONTAINER_IMAGES = {
+    **{
+        # TODO: Re-do manylinux CUDA image tagging scheme to be similar to
+        #       ROCM so we don't have to do this replacement
+        gpu_arch: f"pytorch/manylinux-cuda{gpu_arch.replace('.', '')}"
+        for gpu_arch in CUDA_ARCHES
+    },
+    **{
+        gpu_arch: f"pytorch/manylinux-rocm:{gpu_arch}"
+        for gpu_arch in ROCM_ARCHES
+    },
+    "cpu": "pytorch/manylinux-cpu"
+}
```

Comment:
I guess this is assuming that no cuda version shares its version number with a rocm version? is there a reason we don't just put "cuda" or "rocm" directly into the arch name?

Reply:
It's because I don't want to do a transformation of the value from here to the build scripts, so basically it's to just keep the number as transparent as possible from this point to the build point.

Although a better solution would be to not pass a variable at all and just have it all depend on the container image where the arch version would be automatically baked in as a number into the container image.

Reply:
This approached is outlined in this PR: https://github.com/pytorch/pytorch/pull/45091